controller/controller_server/csi_controller_server.py

```diff
@@ -131,6 +140,75 @@ def CreateVolume(self, request, context):
             context.set_details('an internal exception occurred : {}'.format(ex))
             return csi_pb2.CreateVolumeResponse()
 
+    def _copy_volume_from_snapshot(self, vol, src_snapshot_id, min_vol_size, array_mediator):
```

Comment:
1. maybe copy_to_existing_volume_from_snapshot (we need to distinguish between the transaction overall and specific action inside the transaction)
2. Since the logic is not specific for storage, it can be remain in the controller

Reply:
Renamed to copy_to_existing_volume_from_snapshot

Reply:
Then what is the purpose of `ArrayMediatorAbstract`?
Is there an interest in keeping entire cross-storage logic in 1 class (/file)?

I don't think the location convention is theoretical, and I do see argument sparing as a plus, especially when you didn't provide any cons

Reply:
1. why are the existing `un/map_volume_by_initiators` "ok", but `copy_to_existing_volume_from_snapshot` isn't? what makes it less "storage behavior"?
2. if existing `ArrayMediatorAbstract` is the wrong way to divide the code, then what is the right way? we can't just keep adding all cross-storage code to a single controller class.

Reply:
https://github.com/IBM/ibm-block-csi-driver/pull/483